packages/openapi-generator/src/cli.ts

```diff
@@ -60,23 +60,25 @@ export = class GenerateOpenApiClient extends Command {
 };
 
 /*
-The openapi generator requires a java runtime. In order to get a proper message to the user, we check this here.
+The OpenApi generator requires a Java runtime. In order to get a proper message to the user, we check this here.
  */
-async function checkJavaPresent(): Promise<void> {
-  // TODO: Improve the return values here.
+async function validateJavaRuntime(): Promise<void> {
+  logger.info(
+    'The OpenApi client generation requires a Java runtime. Checking which Java version is installed...'
+  );
+
+  let response;
   try {
-    const response = await execa('java', ['-version']);
-    if (response.exitCode !== 0 || !response.stderr.includes('version')) {
-      logger.error(
-        `Java -version did not return a Java version. Check Java version response: ${response.stderr}`
-      );
-      return Promise.reject();
-    }
-    return Promise.resolve();
+    response = await execa('java', ['-version']);
   } catch (err) {
-    logger.error(
-      `Error in checking the java version. Is Java installed? This is mandatory for the OpenApi client generation. ${err.message}`
+    throw errorWithCause(
+      'Could not invoke `java` command. Probably no Java runtime is installed.',
+      err
+    );
+  }
+  if (response.exitCode !== 0 || !response.stderr.includes('version')) {
+    throw new Error(
+      `Could not validate Java version. Probably Java runtime is corrupt. Version check failed with "${response.stderr}"`
```

Comment:
```suggestion
      `Could not validate Java version. Probably Java runtime is not installed or misconfigured. Version check failed with "${response.stderr}"`
```

Reply:
I think not installed or misconfigured like JAVA_HOME not properly set as env is the most common error.